gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/SchedulerLeaseAlgoHandler.java

```diff
@@ -0,0 +1,144 @@
+      case PREVIOUS_LEASE_VALID:
+        scheduleReminderForTriggerEvent(jobProps, flowGroup, flowName, flowExecutionId, flowActionType, triggerTimeMillis);
+    }
+    return false;
+  }
+
+  /**
+   * This method is used by {@link SchedulerLeaseAlgoHandler.handleNewTriggerEvent} to schedule a reminder for itself to
+   * check on the other participant's progress during pursuing orchestration after the time the lease should expire.
+   * If the previous participant was successful, then no further action is taken otherwise we re-attempt pursuing
+   * orchestration ourselves.
+   * @param flowGroup
+   * @param flowName
+   * @param flowExecutionId
+   * @param flowActionType
+   * @param triggerTimeMillis
+   */
+  protected void scheduleReminderForTriggerEvent(Properties jobProps, String flowGroup, String flowName, String flowExecutionId,
+      SchedulerLeaseDeterminationStore.FlowActionType flowActionType, long triggerTimeMillis) {
+    // Check-in `linger` time after the current timestamp which is "close-enough" to the time the pursuant attempted
+    // the flow action. We also add a small randomization to avoid 'thundering herd' issue
+    String cronExpression = createCronFromDelayPeriod(linger + random.nextInt(staggerUpperBoundSec));
+    jobProps.setProperty(ConfigurationKeys.JOB_SCHEDULE_KEY, cronExpression);
+    // This timestamp is what will be used to identify the particular flow trigger event it's associated with
+    jobProps.setProperty(ConfigurationKeys.SCHEDULER_ORIGINAL_TRIGGER_TIMESTAMP_MILLIS_KEY, String.valueOf(triggerTimeMillis));
+    JobKey key = new JobKey(flowName, flowGroup);
+    Trigger trigger = this.jobScheduler.getTrigger(key, jobProps);
+    try {
+      LOG.info("Attempting to add job reminder to Scheduler Service where job is %s trigger event %s and reminder is at "
+          + "%s.", key, triggerTimeMillis, trigger.getNextFireTime());
+      this.schedulerService.getScheduler().scheduleJob(trigger);
+    } catch (SchedulerException e) {
+      LOG.warn("Failed to add job reminder due to SchedulerException for job %s trigger event %s ", key, triggerTimeMillis, e);
+    }
+    LOG.info(String.format("Scheduled reminder for job %s trigger event %s. Next run: %s.", key, triggerTimeMillis, trigger.getNextFireTime()));
+  }
+
+  /**
+   * These methods should only be called from the Orchestrator or JobScheduler classes as it directly adds jobs to the
+   * Quartz scheduler
+   * @param delayPeriodSeconds
+   * @return
+   */
+  protected static String createCronFromDelayPeriod(long delayPeriodSeconds) {
+    LocalDateTime now = LocalDateTime.now();
+    LocalDateTime delaySecondsLater = now.plus(delayPeriodSeconds, ChronoUnit.SECONDS);
+    // TODO: investigate potentially better way of generating cron expression that does not make it US dependent
+    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("ss mm HH dd MM ? yyyy", Locale.US);
```

Comment:
Does LocalDateTime.now() default to US timezone? Otherwise this could cause issues. I would suggest you use LocalDateTime.now(<timezone>) to ensure consistency in this system across timezones.
Also, I think GaaS scheduler defaults to UTC.

Reply:
I changed `LocaleDateTime.now(UTC)` to ensure the time zone is right but I think the formatter is only putting the timestamp into the cron schedule format, not sure if the locale there matters.